File: CSharp/NetCubeSolver/Benchmarking/BenchmarkResult.cs
namespace NetCubeSolver.Benchmarking;

public class BenchmarkResult
{
    public string ConfigId { get; set; }
    public string ConfigName { get; set; }
    public int Threads { get; set; }
    public int InitialDepth { get; set; }
    public List<long> RunDurationsMs { get; set; } = new();
    public long TotalDurationMs { get; set; }
    public long LongestRunDurationMs { get; set; }
    public long ShortestRunDurationMs { get; set; }
    public long AverageRunDurationMs { get; set; }


    public string GetSummary()
    {
        return $"Benchmark for {ConfigName} Results: Max: {LongestRunDurationMs:n0}, Min: {ShortestRunDurationMs:n0}, Mean: {AverageRunDurationMs:n0} ms";
    }
}

File: Kotlin/app/src/main/kotlin/ca/jwolf/cubesolver/benchmarking/BenchmarkResult.kt
package ca.jwolf.cubesolver.benchmarking

import ca.jwolf.cubesolver.utils.ProgramUtils.Companion.formatWithCommas
import com.fasterxml.jackson.annotation.JsonIgnore

class BenchmarkResult(val configId: String, val configName: String, val threads: Int, val initialDepth: Int,
                      val runDurationsMs: MutableList<Long> = mutableListOf(), var totalDurationMs: Long = 0,
                      var longestRunDurationMs: Long = 0, var shortestRunDurationMs: Long = 0, var averageRunDurationMs: Long = 0 ) {

    @JsonIgnore
    fun getSummary(): String {
        return "Benchmark for $configName Results: Max: ${longestRunDurationMs.formatWithCommas()}, Min: ${shortestRunDurationMs.formatWithCommas()}, Mean: ${averageRunDurationMs.formatWithCommas()} ms"
    }

}